[ads] Followup to "Send HTTP response status for landed events as part of the confirmation token redemption" #36752 (#25523)
brave/brave-core: components/brave_ads/core/internal/account/user_data/README.md

| conversion.envelope[^2]  | {"conversion":[{"envelope":<br>{"alg":"crypto_box_curve25519xsalsa20poly1305",<br>"ciphertext":"3f8a8aeb5e1e4b1a8e7b4e3f8a8aeb5e1e4b1a8e7b4e3f8a8aeb5e1e4b1a8e7b",<br>"epk":"d1f2e3d4c5b6a79887a6b5c4d3e2f1d1f2e3d4c5b6a79887a6",<br>"nonce":"a1b2c3d4e5f6a7b8c9d0e1f2a3b4c5d6"}}]}  | Allow advertisers to affirm that conversions are legitimate by wrapping an identifier with an additional layer of encryption to protect its integrity. Also, see https://ads-help.brave.com/campaign-performance/reporting#verifiable-ad-conversions-vac.  |
| createdAtTimestamp  | {"createdAtTimestamp":"2020-11-18T12:00:00.000Z"}  | [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp with fixed values for minutes, seconds, milliseconds and timezone.  |
| diagnosticId[^1][^2]  | {"diagnosticId":"c1298fde-7fdb-401f-a3ce-0b58fe86e6e2"}  | Diagnostic id from [brave://rewards-internals](brave://rewards-internals) to assist in troubleshooting issues. This is only included if it has been set manually by the user, for example, at the request of Brave Support in order to troubleshoot a problem.  |

> | httpResponseStatus  | {"httpResponseStatus":"4xx"}  | Indicates the landing page's HTTP status code.  |

Reviewer: ```suggestion
| httpResponseStatus  | {"httpResponseStatus":"404"}  | Indicates the landing page's HTTP status code.  |
```

> | httpResponseStatus  | {"httpResponseStatus":"4xx"}  | Indicates the landing page's HTTP status code, where privacy-preserving status codes are limited to include only the status code class.  |

Reviewer: I don't think we need to say more than this:
```suggestion
| httpResponseStatus  | {"httpResponseStatus":"4xx"}  | Indicates the landing page's HTTP status code.  |
```
Limiting the data to just the codes we need (and using `4xx` or `5xx` for the others) is data minimization, but I wouldn't call it a privacy-preserving technique.

> | httpResponseStatus  | {"httpResponseStatus":"4xx"}  | Indicates the landing page's HTTP status code, where privacy-preserving status codes are limited to include only the status code class.  |

Reviewer 2: ```suggestion
| httpResponseStatus  | {"httpResponseStatus":"4xx"}  | Indicates the landing page's HTTP response status code class (full status code not sent for privacy reasons). |
```

> | httpResponseStatus  | {"httpResponseStatus":"4xx"}  | Indicates the landing page's HTTP status code, where privacy-preserving status codes are limited to include only the status code class.  |

Reviewer 2: Looking at https://github.com/brave/reviews/issues/1553#issuecomment-2334753079, it seems like we aren't actually sending just the status code class. In some cases, we're sending the full status code. If that's the case, then I agree with @fmarier's suggested text.
